pyband/wallet/signer.py

```diff
@@ -0,0 +1,84 @@
+from abc import abstractmethod
+
+from ecdsa.der import remove_sequence, UnexpectedDER, remove_integer
+
+from .address import Address
+from .private_key import PrivateKey
+from .public_key import PublicKey
+from ..cosmos_app import CosmosApp
+from ..utils import bip44_to_list
+
+
+class Signer:
+    def __init__(self):
+        self.public_key = self.get_public_key()
+        self.address = self.get_address()
+
+    @abstractmethod
+    def get_public_key(self) -> PublicKey:
+        """Gets public key associated with the signer.
+
+        Returns:
+            A PublicKey instance.
+        """
+        pass
+
+    @abstractmethod
+    def get_address(self) -> Address:
+        """Gets address associated with the signer.
+
+        Returns:
+            An Address instance.
+        """
+        pass
+
+    @abstractmethod
+    def sign(self, msg: bytes) -> bytes:
+        """Signs a message with the signer.
+
+        Args:
+            msg: The message to sign in bytes.
+
+        Returns:
+            Signature in bytes.
+        """
+        pass
+
+
+class PrivateKeySigner(Signer):
+    def __init__(self, private_key: PrivateKey):
+        super().__init__()
```

Comment:
it will cause an error because "super" calls get_public_key which requires private_key but private_key is still null at that point.

Reply:
@taobun forgot `super().__init__()` needs to be at the end, position should stay where it is